letsencrypt-auto-source/letsencrypt-auto

```diff
@@ -898,13 +898,18 @@ if [ "$1" = "--le-auto-phase2" ]; then
   SetPrevBootstrapVersion
 
   INSTALLED_VERSION="none"
-  if [ -d "$VENV_PATH" ]; then
+  if [ -d "$VENV_PATH" -o -d "$OLD_VENV_PATH" ]; then
     # If the selected Bootstrap function isn't a noop and it differs from the
     # previously used version
     if [ -n "$BOOTSTRAP_VERSION" -a "$BOOTSTRAP_VERSION" != "$PREV_BOOTSTRAP_VERSION" ]; then
       # if non-interactive mode or stdin and stdout are connected to a terminal
       if [ \( "$NONINTERACTIVE" = 1 \) -o \( \( -t 0 \) -a \( -t 1 \) \) ]; then
-        rm -rf "$VENV_PATH"
+        if [ -d "$VENV_PATH" ]; then
```

Comment:
If, for some reason both, the `$VENV_PATH` and `$OLD_VENV_PATH` exist here, we're never able to get to a sane state.

Reply:
While this case is pretty unlikely, there are a few ways it could happen so it's definitely worth thinking about. Nice catch.

My understanding of what currently happens in this case is we hit this line, see that `$VENV_PATH` exists, delete it, and then rerun the script. In this next run, we hit "phase 1", see that `$OLD_VENV_PATH` exists [here](https://github.com/certbot/certbot/blob/v0.20.0/certbot-auto#L1324), decide not to bootstrap, and run the script again with `--le-auto-phase2` to re-enter "phase 2".

In "phase 2", we check if `$VENV_PATH` exists (which it doesn't), but `$OLD_VENV_PATH` exists, so we delete it, symlink it, and run this script yet again, however, this time, when hitting "phase 1", neither installation location exists so we rebootstrap and everything works.

This is pretty ugly and excessive though. I'll change this up to delete any installation location that exists here, but I'm also going to change this up to check `$OLD_VENV_PATH` more deeply (by specifically looking for `$OLD_VENV_PATH/bin/letsencrypt` as I think one of the cases where both exist is `~/.local/share/letsencrypt` is a location made by the user and not `certbot-auto`.

If you disagree, please let me know!